python/feldera/sql_schema.py

```diff
@@ -0,0 +1,15 @@
+from typing import Mapping
+
+
+class SQLSchema:
+    schema: Mapping[str, str]
```

Comment:
Schema should capture the nullability of columns, primary key constraints, lateness, and watermarks. Perhaps it should be modeled after the Rust `Relation` type. We can add that in future PRs.